File: example/src/main/java/ru/virgil/example/security/SecurityUserDetailsRepository.java
package ru.virgil.example.security;

import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

import java.util.Optional;
import java.util.UUID;

@Repository
public interface SecurityUserDetailsRepository extends JpaRepository<SecurityUserDetails, UUID> {

    Optional<SecurityUserDetails> findByFirebaseUserId(String firebaseUserId);

}

File: example/src/main/java/ru/virgil/example/security/SecurityUserDetailsRepository.kt
package ru.virgil.example.security

import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository
import java.util.*

@Repository
interface SecurityUserDetailsRepository : JpaRepository<SecurityUserDetails, UUID> {

    fun findByFirebaseUserId(firebaseUserId: String): Optional<SecurityUserDetails>
}
